File: App/Frontend/frontend/lib/models/postProblemType.dart
import 'package:Frontend/models/enums.dart';

class PostProblemType {
  int id;
  int postId;
  int problemType;

  PostProblemType(this.postId, this.problemType);

  PostProblemType.fromJSON(Map<String, dynamic> json){
    this.id = json["id"];
    this.postId = json["postId"];
    this.problemType = json["problemType"];
  }

  Map<String, dynamic> toJSON(){
    var map = Map<String, dynamic>();

    map["id"] = id;
    map["postId"] = postId;
    map["problemType"] = problemType;
    return map;
  }
}

File: App/Backend/Backend/Backend/Models/PostProblemType.cs
using Backend.Models.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Backend.Models
{
    public class PostProblemType
    {
        public int Id { get; set; }
        public int PostId { get; set; }
        public ProblemType ProblemType { get; set; }
    }
}
